fastai/learner.py

```diff
@@ -97,22 +97,22 @@ def before_epoch(self):
 class Learner(GetAttr):
     _default='model'
     def __init__(self,
-                 dls,  # `DataLoaders` containing data for each dataset needed for `model`
-                 model:callable, #  The model to train or use for inference
-                 loss_func:callable|None=None,  # Loss function for training
-                 opt_func=Adam,  # Optimisation function for training
-                 lr=defaults.lr,  # Learning rate
-                 splitter:callable=trainable_params,  # Used to split parameters into layer groups
-                 cbs=None,  # Callbacks
-                 metrics=None,  # Printed after each epoch
-                 path=None,  # Parent directory to save, load, and export models
-                 model_dir='models',  # Subdirectory to save and load models
-                 wd=None,  # Weight decay
-                 wd_bn_bias=False,  # Apply weight decay to batchnorm bias params?
-                 train_bn=True,  # Always train batchnorm layers?
-                 moms=(0.95,0.85,0.95),  # Momentum
-                 default_cbs:bool=True  # Include default callbacks?
-                ):
+        dls:DataLoaders, # `DataLoaders` containing fastai or PyTorch `DataLoader`s
+        model:callable|nn.Module, # PyTorch model for training or inference
+        loss_func:nn.Module|None=None, # Loss function. Defaults to `dls` loss
+        opt_func:Optimizer|OptimWrapper=Adam, # Optimization function for training
+        lr:float|slice=defaults.lr, # Default learning rate
+        splitter:callable=trainable_params, # Split model into parameter groups. Defaults to one parameter group
+        cbs:Callback|list|None=None, # `Callback`s to add to `Learner`
+        metrics:callable|list|None=None, # `Metric`s to calculate on validation set
+        path:str|Path|None=None, # Parent directory to save, load, and export models. Defaults to `dls` `path`
+        model_dir:str|Path='models', # Subdirectory to save and load models
+        wd:float|None=None, # Default weight decay
+        wd_bn_bias:bool=False, # Apply weight decay to normalization and bias parameters
+        train_bn:bool=True, # Train frozen normalization layers
+        moms:tuple[float,...]=(0.95,0.85,0.95), # Default momentum for schedulers
```

Comment:
We don't use generic types in any fast.ai lib

Reply:
fastai has `moms:tuple=(0.95,0.85,0.95)` in multiple places throughout the library. Prefer that formatting?

Reply:
Yes exactly - just remove the generic constraint and leave it as `tuple`.